### ISSUE TITLE
Question: Is Project Still Maintained

### ISSUE THREAD

**Issue by b1tzer0** [Azure/elastic-db-tools#221]
Is this project still being maintained?  It is still referenced in the Microsoft Docs.

I want to use this in a new project, but if it is not being maintained then I might look elsewhere for a multi-tenant solution.
Any recommendations?

**Comment by scoriani** [comment]
Hey @b1tzer0, while we don't have short term plans on improving this project, scale out and multi-tenant solutions are still very interesting areas for us, and it would be great if we could follow up offline on your specific requirements and use cases. If you're interested, please drop me an email (scoriani at microsoft.com) and I'll set up a quick call.

**Comment by christophwille** [comment]
Hi @scoriani - one item that immediately comes to mind is #220 with the related PR #217 (full .NET 6.0 target + Ms.D.SqlClient for modern authN methods)

**Comment by scoriani** [comment]
Thanks @christophwille, those are quite straightforward. I was more interested in the intersections between scaling out a database backend while implementing a multi-tenant solution. There are multiple angles to consider, from developer productivity to performance to security/isolation that play a critical role and wanted to double check if we're not missing anything important in addition to what has already been covered in guides like https://docs.microsoft.com/en-us/azure/azure-sql/database/saas-tenancy-app-design-patterns

**Comment by vigouredelaruse** [comment]
just noticing an multitenancy pattern on microsoft impedence mismatch between
- asp.net/core - entity framework core, that has architectural features that defeat multitenancy out of the box; no dynamic auth provider switching, no dynamic dbconnection switching for example
- azure client sdks that are cloud multitenancy enabled out of the box like the elastic tools library that has no important commits in years

for anyone observing microsoft employees observing this situation one can't help but point out a speculative outflow of implementers building multitenant solutions

**Comment by scoriani** [comment]
Hi @vigouredelaruse, would you mind clarifying a bit your comment above as I'm not sure I understand your point? On the technologies you mentioned, there are Microsoft 1st party and 3rd party ISVs that have built multitenant solutions with millions of tenants and databases but, if there's any valuable feedback you can provide that will be very appreciated.

**Comment by vigouredelaruse** [comment]
howdy @scoriani; not to have a spinal tap moment with the napkin here but take a look at some objects that would be nice to have in context in an asp.net family instance

![image](https://user-images.githubusercontent.com/31052867/165381123-7b6a34a0-18d3-4265-9de9-1f64d5549e94.png)

it's incontrovertible that one can build a multitenant implementation using the technologies in scope of this discussion. the question that started this thread though implies a question: can one plan for robust support to further such efforts using the technologies in scope of this discussion

this https://docs.microsoft.com/nb-no/ef/core/miscellaneous/multitenancy tells you how to do it

what's missing is some handholding from a suite of boilerplate components that you can install as project starter templates in visual studio

hopefully my spinal tap diagram gives an idea of what i'm having to cobble together right now to fulfill a research grant from the canadian federal govt - i need a cms where i can deploy a gis system that can easily interoperate with the tools used in the gis community and the official visual studio template to start such a thing just isn't there

 https://dev.azure.com/electricrucible/the%20horseless%20newspaper/_wiki/wikis/the%20horseless%20wiki/120/this-functional-specification

**Comment by scoriani** [comment]
Thanks @vigouredelaruse, much clearer now. This s bit outside of just the database layer (which is where I work) but I'll bring the feedback back to Visual Studio and .NET team to see if we can make it better.

**Comment by vigouredelaruse** [comment]
well yes there are some serious cross cutting concerns that prescribe the current situation. if anyone is interested i'm sure enough people are interested in database sharding and the surrounding issue to get a real nice diagram going :)

**Comment by JonPSmith** [comment]
Hi @vigouredelaruse,

I saw your comment and I though you might be interested in an article I just released called [How to take an ASP.NET Core web site “Down for maintenance”](https://www.thereformedprogrammer.net/how-to-take-an-asp-net-core-web-site-down-for-maintenance/#3-using-the-isetremovestatus-service-to-set-remove-a-down-state) where I describe a way to divert tenant users while the tenant is being split / merged by an admin user. 

This describes how to create this feature in an ASP.NET Core app, with a emphasis on not allowing the feature from effecting the application's performance. I needed this feature for my [AuthPermissions.AspNetCore](https://github.com/JonPSmith/AuthPermissions.AspNetCore) library.

PS. You might also like to look at the previous article which talks about [how to form the correct connection string](https://www.thereformedprogrammer.net/how-to-turn-an-asp-net-core-appsettings-json-file-into-a-fast-read-database/#setting-the-scene-why-did-i-use-an-appsettings-json-file-as-a-database) in a multi-tenant application using sharding. Again this is very fast (and secure).
